src/Uno.UI.Runtime.Skia.Wpf/WPF/WpfDragDropExtension.cs

```diff
@@ -0,0 +1,279 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.IO;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using Windows.ApplicationModel.DataTransfer;
+using Windows.ApplicationModel.DataTransfer.DragDrop;
+using Windows.ApplicationModel.DataTransfer.DragDrop.Core;
+using Windows.Storage;
+using Windows.Storage.Streams;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Input;
+using Uno.Extensions;
+using Uno.Logging;
+using DragEventArgs = System.Windows.DragEventArgs;
+using Point = Windows.Foundation.Point;
+using UIElement = Windows.UI.Xaml.UIElement;
+using Window = System.Windows.Window;
+
+namespace Uno.UI.Skia.Platform
+{
+	public class WpfDragDropExtension : IDragDropExtension
+	{
+		private readonly long _fakePointerId = Pointer.CreateUniqueIdForUnknownPointer();
+		private readonly DragDropManager _manager;
+
+		public WpfDragDropExtension(object owner)
+		{
+			_manager = (DragDropManager)owner;
+
+			var host = WpfHost.Current;
+
+			host.AllowDrop = true;
+
+			host.DragEnter += OnHostDragEnter;
+			host.DragOver += OnHostDragOver;
+			host.DragLeave += OnHostDragLeave;
+			host.Drop += OnHostDrop;
+		}
+
+		private void OnHostDragEnter(object sender, DragEventArgs e)
+		{
+			var src = new DragEventSource(_fakePointerId, e);
+			var data = ToDataPackage(e.Data);
+			var allowedOperations = ToDataPackageOperation(e.AllowedEffects);
+			var info = new CoreDragInfo(src, data.GetView(), allowedOperations);
+
+			CoreDragDropManager.GetForCurrentView()?.DragStarted(info);
+
+			// Note: No needs to _manager.ProcessMove, the DragStarted will actually have the same effect
+		}
+
+		private void OnHostDragOver(object sender, DragEventArgs e)
+			=> e.Effects = ToDropEffects(_manager.ProcessMoved(new DragEventSource(_fakePointerId, e)));
+
+		private void OnHostDragLeave(object sender, DragEventArgs e)
+			=> e.Effects = ToDropEffects(_manager.ProcessAborted(new DragEventSource(_fakePointerId, e)));
+
+		private void OnHostDrop(object sender, DragEventArgs e)
+			=> e.Effects = ToDropEffects(_manager.ProcessDropped(new DragEventSource(_fakePointerId, e)));
+
+		public void StartNativeDrag(CoreDragInfo info)
+			=> WpfHost.Current.Dispatcher.InvokeAsync(async () =>
+			{
+				try
+				{
+					var data = await ToDataObject(info.Data, CancellationToken.None);
+					var effects = ToDropEffects(info.AllowedOperations);
+
+					DragDrop.DoDragDrop(WpfHost.Current, data, effects);
```

Comment:
How is Uno notified when the host completes a drop? How is DropCompleted raised? Maybe that comes through the window events (OnHostDrop) again?

Reply:
Edit: Agree to address this later as needed. (assuming I'm understanding correctly anyway).